instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.nightly.logs
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.log.Logger
+import com.datadog.android.nightly.rules.NightlyTestRule
+import com.datadog.android.nightly.utils.initializeSdk
+import com.datadog.android.nightly.utils.measureLoggerInitialize
+import com.datadog.android.rum.GlobalRum
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class LoggerBuilderE2ETests {
+
+    @get:Rule
+    val forge = ForgeRule()
+
+    @get:Rule
+    val nightlyTestRule = NightlyTestRule()
+
+    lateinit var logger: Logger
+
+    @Before
+    fun setUp() {
+        initializeSdk(InstrumentationRegistry.getInstrumentation().targetContext)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setSampleRate(Float): Builder
+     */
+    @Test
+    fun logs_logger_builder_sample_all_in() {
+        val testMethodName = "logs_logger_builder_sample_all_in"
+        measureLoggerInitialize {
+            logger = Logger.Builder().setSampleRate(1f).build()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setSampleRate(Float): Builder
+     */
+    @Test
+    fun logs_logger_builder_sample_all_out() {
```

Comment:
Maybe we need one or more to test what happens when we use a value between 0 and 1, e.g 75%

Reply:
Yes, I also thought of that but then problem is how would you assert that test case as you are not sure if this event is passing your sampling or not ?

Reply:
You send `n` logs and assert that the number of logs with the matching tag you receive is roughly equal to `n⨉75%`